ts/parts-3d/Column.ts

```diff
@@ -500,27 +513,56 @@ if (seriesTypes.columnrange) {
 
 wrap(Series.prototype, 'alignDataLabel', function (
     this: Highcharts.Series,
-    proceed: Function
+    proceed: Function,
+    point: Highcharts.ColumnPoint,
+    dataLabel: Highcharts.SVGElement,
+    options: Highcharts.DataLabelsOptionsObject,
+    alignTo: Highcharts.BBoxObject
 ): void {
+    const chart = this.chart;
+
+    // In 3D we need to pass point.outsidePlot option to the justifyDataLabel
+    // method for disabling justifying dataLabels in columns outside plot
+    options.outside3dPlot = point.outside3dPlot;
 
     // Only do this for 3D columns and it's derived series
     if (
-        this.chart.is3d() &&
+        chart.is3d() &&
         this instanceof seriesTypes.column
     ) {
-        var series = this as Highcharts.ColumnSeries,
-            chart = series.chart;
-
-        var args = arguments,
-            alignTo = args[4],
-            point = args[1];
-
-        var pos = ({ x: alignTo.x, y: alignTo.y, z: series.z });
+        const series = this as Highcharts.ColumnSeries,
+            seriesOptions: Highcharts.ColumnSeriesOptions = series.options,
+            inside = pick(options.inside, !!series.options.stacking),
+            options3d = (chart.options.chart as any).options3d;
+
+        let dLPosition = {
+            x: alignTo.x + point.pointWidth / 2,
+            y: alignTo.y,
+            z: series.z + (seriesOptions as any).depth / 2
+        };
+
+        if (chart.inverted) {
+            // dLPosition.y += (point.shapeArgs as any).width / 2;
+            // Inside dataLabels are positioned according to above
+            // logic and there is no need to position them using
+            // non-3D algorighm (that use alignTo.width)
+            if (inside) {
+                alignTo.width = 0;
+                dLPosition.x += (point.shapeArgs as any).height / 2;
+            }
+            // When chart is upside down
+            // (alpha angle between 180 and 360 degrees)
+            // it is needed to add column width to calculated value.
+            if (options3d.alpha >= 90 && options3d.alpha <= 270) {
```

Comment:
I am unable to find in code normalization to 0-360 degree range for alpha. Is rotating the chart for x + 360*n covered? or not really supported anywhere and this place shouldn't be different?

Reply:
Thanks! I will add it in the next commit. To be honest, I found the same issue just an hour ago. We should cover this normalisation for every use case, not only for dataLabels.